cmd/cli/serve/timeout_test.go

```diff
@@ -0,0 +1,72 @@
+//go:build unit || !integration
+
+package serve_test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/requester/publicapi"
+)
+
+var (
+	noTimeout          = model.NoJobTimeout
+	nonZeroTimeout     = 30 * time.Second
+	halfNonZeroTimeout = nonZeroTimeout / 2
+)
+
+func (s *ServeSuite) TestNoTimeoutSetOrApplied() {
+	docker.MustHaveDocker(s.T())
+
+	cases := []struct {
+		configuredMax    *time.Duration
+		timeoutSpecified *time.Duration
+		timeoutApplied   time.Duration
+		stateExpected    model.JobStateType
+	}{
+		{configuredMax: nil, timeoutSpecified: nil, timeoutApplied: model.NoJobTimeout, stateExpected: model.JobStateCompleted},
+		{configuredMax: nil, timeoutSpecified: &nonZeroTimeout, timeoutApplied: nonZeroTimeout, stateExpected: model.JobStateCompleted},
+		{configuredMax: &nonZeroTimeout, timeoutSpecified: nil, timeoutApplied: nonZeroTimeout, stateExpected: model.JobStateCompleted},
+		{configuredMax: &nonZeroTimeout, timeoutSpecified: &halfNonZeroTimeout, timeoutApplied: halfNonZeroTimeout, stateExpected: model.JobStateCompleted},
+		{configuredMax: &nonZeroTimeout, timeoutSpecified: &noTimeout, timeoutApplied: noTimeout, stateExpected: model.JobStateError},
+	}
+
+	for _, tc := range cases {
+		name := fmt.Sprintf(
+			"job in %s has timeout %s after configuring %s and specifying %s",
+			tc.stateExpected,
+			tc.timeoutApplied,
+			tc.configuredMax,
+			tc.timeoutSpecified,
+		)
+
+		s.Run(name, func() {
+			args := []string{}
+			if tc.configuredMax != nil {
+				args = append(args, "--max-timeout", tc.configuredMax.String())
+			}
+
+			port, err := s.serve(args...)
+			s.Require().NoError(err)
+
+			client := publicapi.NewRequesterAPIClient("localhost", port)
+			s.Require().NoError(publicapi.WaitForHealthy(s.ctx, client))
+
+			job, err := model.NewJobWithSaneProductionDefaults()
```

Comment:
Me too! At the very least, I'd like more tests to start using WASM jobs because they run on more platforms in CI than Docker.

How would you feel if I:

1. Changed the `NewJobWithSaneProductionDefaults` to a real fully valid WASM example?
2. Removed the use of the function entirely?

Reply:
I have gone for the secret third option which is just changing this case. I figure it will be less burdensome for your work if we do a change like that after it's merged.